dot/parachain/availability-store/availability_store.go

```diff
@@ -4,169 +4,500 @@
 package availabilitystore
 
 import (
+	"bytes"
 	"context"
 	"encoding/binary"
-	"encoding/json"
 	"errors"
 	"fmt"
+	"time"
 
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
 	"github.com/ChainSafe/gossamer/internal/database"
 	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/erasure"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
 var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-availability-store"))
 
 const (
-	avaliableDataPrefix = "available"
+	availableDataPrefix = "available"
 	chunkPrefix         = "chunk"
 	metaPrefix          = "meta"
 	unfinalizedPrefix   = "unfinalized"
 	pruneByTimePrefix   = "prune_by_time"
+
+	// Unavailable blocks are kept for 1 hour.
+	KeepUnavilableFor = time.Hour
+
+	// Finalized data is kept for 25 hours.
+	KeepFinalizedFor = time.Hour * 25
+
+	// The pruning interval.
+	PruningInterval = time.Minute * 5
 )
 
+// BETimestamp is a unix time wrapper with big-endian encoding
+type Timestamp uint64
+
+// ToBEBytes returns the big-endian encoding of the timestamp
+func (b Timestamp) ToBEBytes() []byte {
+	res := make([]byte, 8)
+	binary.BigEndian.PutUint64(res, uint64(b))
+	return res
+}
+
+type SubsystemClock struct{}
+
+func (sc *SubsystemClock) Now() Timestamp {
+	return Timestamp(time.Now().Unix())
+}
+
+// PruningConfig Struct holding pruning timing configuration.
+// The only purpose of this structure is to use different timing
+// configurations in production and in testing.
+type PruningConfig struct {
+	keepUnavailableFor time.Duration
+	keepFinalizedFor   time.Duration
+	pruningInterval    time.Duration
+}
+
+var DefaultPruningConfig = PruningConfig{
+	keepUnavailableFor: KeepUnavilableFor,
+	keepFinalizedFor:   KeepFinalizedFor,
+	pruningInterval:    PruningInterval,
+}
+
 // AvailabilityStoreSubsystem is the struct that holds subsystem data for the availability store
 type AvailabilityStoreSubsystem struct {
```

Comment:
does not get used outside this package, let's unexport this.

Reply:
Updated.